---
title: "Data Science project: thyroid specific antigens"
subtitle: "The file with all of our Code"
author: "Camila Vacas, Line Weiß, Carina Keßler"
date: "19/7/2021"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Order of this file:
1. All the Text from the report
2. All of our Code (see as well on Github), ordered in chunks
3. Appendix


# Introduction

Tissue restricted antigens (TRA) are those genes which are expressed in at most five out of fourty-five tissues, and additionally in the lymph system, e.g. in the medullary thymic epithelial cells (mTECs), where the negative selection of the T-cells takes place (Kyewske and Klein, 2006). A defective regulation of TRA expression in mTECs can lead to autoimmune diseases, caused by the presence of T-cells reacting to a self-antigen. In most patients of e.g. the Hashimoto's or Grave's disease, the most common autoimmune disorders concerning the thyroid, are antibodies present against the thyroglobulin or the thyroid peroxidase. 
Due to this negative selection, the immune system cannot react properly to tumor cells which express tumor associated antigens (TAA), like differentiation or onco-fetal antigens (Kyewske and Klein,2006). The tumor escapes the immune system and can grow. Radiation has a positive impact on tumor development as well, especially in thyroid cancer: the thyroid is higly sensitive to radiation exposure and it is likely to develop long-term effects. Though females tend to develope more often thyroid cancer than males, youth is a greater risk factor. (Albi, 2017)

Papillary thyroid cancer (PTC) represents 80 % of thyroid tumors.Usually, there is one gene mutated in the MAPK signal cascade, the most frequently affected genes are *braf* (point mutation, mostly V600E) and *ret* (RET/PTC rearrangement).

We were working with 14 samples of thyroid tissues. Five of them were extracted from patients, who were exposed to radiation and developed PTC, another four came from patients who were not exposed to radiation and developed sporadic PTC and the last five samples were from patients with healthy thyroid tissue. The gene expression analysis was performed using the gene chip hgu133plus2hsenstcdf.

Our first step of the quality control was the single chip control, and we decided to keep all of our chips. The normalization we did afterwards is necessary to reduce the impact of the usage of different amounts of fluorescent dye and it made the data comparable. In the RNA degradation plots were some slopes deviating from the group pattern, which indicates that a lot of RNA was degrated. Even though there were some irregularities in this plot, we decided to reject none of our chips, because the rest of the quality control seemed to be fine.


![Caption for the picture 1](/Users/Line/Documents/tra_project/boxplot_thyroid_rawdata){width=50%}
![Caption for the picture 2](/Users/Line/Documents/tra_project/boxplot_thyroid_vsnrma_normalized){width=50%}
Figure 0: gene expression in the thyroid tissue of 14 patients, who were either exposed to radion, not exposed to radiation or healthy, before and after normalization.The boxplots show the distribution of the expression of all genes per chip before and after normalization. The goal is to bring the quantiles more or less into line, which makes the data comparable. 



In the next steps we focused on the TRAs in the thyroid (our genes of interest), and extracted them by using five TRA lists (from human and mouse samples). We found 360 TRAs of thyroid tissue on our microarrays (see appendix) and created a matrix containig our genes of interest and their expression data. This matrix was the basis for our following analyses.

We wanted to find out which of these thyroid specific genes are differentially expressed in low-dose radiation-induced PTC, sporadic PTC and healthy thyroid tissue and how the gene expression differs between the three groups. 



# Main Part
  
We started our project by taking a closer look at the data. We found out, that our genes of interest are spread over all human chromosomes, except for the Y chromosome. Unfortunately we cannot say on which chromosomes the most thyroid specific TRAs can be found, due to the missing information of chromosome localisation of almost half of our genes.   



We also created a heatmap to get an overview of the gene expression of the thyroid specific antigens (see below).



We decided to perform a principal component analysis (PCA) to detect any patterns. We coloured the different spots according to the tissue sample the microarray was prepared with. The five healthy samples and the five radiation exposed samples formed both a clear individual cluster. Two of the samples of the sick patients, who were not exposed to radiation, formed a seperated cluster, while the other two seemed to be more similiar to the radiation exposed samples.

                       
![PCA 14 chips](/Users/Line/Documents/tra_project/thyroid_cancer_sessions/pca_genes.jpg){width=50%}
  
Figure 1.2: PCA of the 14 chips colored according to to tissue origin.The healthy tissue samples form a clear cluster, while the sick samples did not. 



To find out, wether those were just outliers, we repeated the PCA with 30 chips, 10 chips in each group. This time it was clear, that the healthy samples formed one cluster and the sick samples another one. 



To prove this, we performed k-means clustering and used the elbow method to determine the best number of clusters, which was two. 



To investigate, which of the 360 TRA's showed a significant change in gene expression upon the three different groups, we performed an ANOVA test for each gene. We found 153 significantly different expressed genes (see appendix) and stored them in a new matrix.



The posthoc test of each of the significantly different expressed genes showed us how the acccording gene was expressed in the three groups and between which of the groups was a significant difference. For that we used the TukeyHSD function to perform the tukey honestly significant differnce posthoc test. It enables us to compare the differences of the mean gene expression between our three groups. It helped us with the visualisation and analysis of the ANOVA test as well.   



Due to the results of the PCA, which showed us that the gene expression of radiation exposed and non-exposed tissue did not differ as much as we thought it would, we decided to concentrate ourselfs on the cancer tissue further. We performed another PCA and a k-means clustering to investigate whether there is another pattern within these groups.



We found that the chips could not be seperated into two groups according to their radiation exposure. 
The elbow method,we used to define the perfect number of clusters, revealed, that the cancer tissues should be divided into three groups, illustrated below. We assumed that these could be three different PTC subtypes.



To look into this, we examined the gene expression pattern of these three cancer subtypes. We performed another ANOVA test, defining the genes, that are significantly different expressed in our subtypes. We found 134 genes (see appendix). 



We wanted to design a regression model to predict the cancer subtype through the expression values of the significant genes. Since 134 variables are to many to work with, we performed another posthoc TukeyHSD test and selected the genes that are differentially expressed in all three subtypes (see appendix). With these eleven genes we trained our multinominal logistic regression model with 14 chips and tested it on another six chips.



# Results:

We identified 560 TRAs in the thyroid, but only 360 were represented on the microarray-type we used. Out of these 360 TRAs 153 were significantly different expressed between the three groups healthy, exposed and not-exposed to radiation. 
Through PCA we came across that there seemed to be no difference in the gene expression between the two carcinogenic tissue types. However, there was a clear difference between the gene expression in the healthy patients compared to the sick ones. We focused on the sick samples further and found three clear groups that differ in gene expression and suspected that we had discovered three subtypes of PTC. 
Moreover, we decided to create a regression model that would enable us to determine the subtype according to the gene expression. Therefore, we identified the 134 genes that were significantly different expressed between at least two of the subtypes using an ANOVA test. But since 134 genes were too many we narrowed them down to eleven genes, which were significantly different expressed between all three subtypes. Based on this eleven genes we were able to design a regression model with an accuracy of 100 %. 



# Discussion:

We are aware that the number of chips we used is to little to get valid results. Although we could not find any clear differences between the radiation-exposed and not-exposed samples, an examination with more than the 30 chips, could come to another conclusion. 
Furthermore, we looked into the known variants of PTC and found out that there are three, that occur the most: 
the conventional variant, the follicular variant and the tall cell variant (LLyod et al., 2011). Even though there are many other less common subtypes as the three we mentioned before, we suspect that the three groups we discovered correspond to these variants. To prove this further analyses would be necessary.
Additionally, the accuracy of the regression model seemed to high at first. But regarding the criteria for selecting the eleven genes the model is based on, it seems logical to get an accuracy of 100 %: All of our genes show a unique expression level in each subtype, so the model prediction has to be unambiguously.

During our work on this report we discovered that certain analyses are based on coincidence, which prevented us from recreating the same results. After looking into it, we found out that the expression values of our TRAs as well as the kmeans clustering results differed a little bit after every run of our code. We think that the change in the expression values is due to the normalization process, because it is a method based on coincidence, as well as the kmeans clustering method. This is why we decided to load those two variables into the RMarkdown, to present the results we found during our project. We are aware that this limits the validity of our results.



# Literature: 

TRA datas:

GTEX 2015, Human genomics (2015). The Genotype-Tissue Expression (GTEx) pilot analysis: multitissue gene regulation in humans. GTEx Consortium. Science 348(6235),648-60. 

Lattin, J. E., Schroder, K., Su, A. I., Walker, J. R., Zhang, J., Wiltshire, T., Saijo, K., Glass, C. K., Hume, D. A., Kellie, S., Sweet, M. J. (2008). Expression analysis of G Protein-Coupled Receptors in mouse macrophages. Immunome Res. 29,4-5.

Roth, R.B., Hevezi, P. Lee, J., Willhite, D., Lechner, S. M., Foster, A. C., Zlotnik, A. (2006). Gene expression analyses reveal molecular relationships among 20 regions of the human CNS. Neurogenetics 2,67-80. 

Su, A. I., Cooke, M. P., Ching, K. A., Haka, Y., Walker, J. R., Wiltshire, T., Orth, A. P., Vega, R. G., Sapinoso, L. M., Moqrich, A., Patapoutian, A., Hampton, G. A., Schultz, P. G., Hogenesch, J. B. (2002). Large-scale analysis of the human and mouse transcriptomes. Proc Natl Acad Sci U S A 99(7), 4465-70.

Su, A. I., Wiltshire, T., Batalov, S., Lap, H., Ching, K. A., Block, D., Zhang, J., Soden, R., Hayakawa, M., Kreiman, G., Cooke, M. P., Walker, J. R., Hogenesch, J. B. (2004). A gene atlas of the mouse and human protein-encoding transcriptomes. Proc Natl Acad Sci U S A 101(16),6062-7.

Uhlén, M., Fagerberg, L., Hallström, B.M., Lindskog, C., Oksvold, P., Mardinogl, A., Sivertsson, A., Kampf, C., Sjöstedt, E., Asplund, A., Olsson, I.M., Edlund, K., Lundberg, E., Navani, S., Szigyart, C.A.K., Odeberg, J., Djureinovic, D., Takanen, J.O., Hober, S., Alm, T., Edqvist, P.H., Berling, H., Tegel, H., Mulder, J., Rockberg, J., Nilsson, P., Schwenk, J.M., Hamsten, M., Feilitzen, K., Forsberg, M., Persson, L., Johansson, F., Zwahlen, M., Haijne, G., Nielsen, J. and Pontén, F. (2015). Proteomics. Tissue-based map of the human proteome. Science. 23;347(6220):1260419.

Others: 

Albi, E., Cataldi, S., Lazzarini, A., Codini, M., Beccari, T., Ambesi-Impiombato., F. S., Curcio, F (2017). Radaition and Thyroid Cancer. Int. J. Mol. Sci. 18(5), 911

Dinkelacker, M. (2019). Chromosomal clustering of tissue restricted antigens, Dissertation, University Heidelberg, Germany.

Dinkelacker, M. (2007). A database of genes that are expressed in a tissue-restricted manner to analyse promiscous gene expression in medullary thymic epithelial cells. Diplomarbeit, Albert-Ludwigs-Universitaet, Freiburg, Germany.

Kyewski, B. and Klein, L. (2006). A Central Role for Central Tolerance. Annu. Rev. Immunol. 24, 571-606.

Lloyd, R. V., Buehler, D. and Khanfshar, E. (2011). Papillary Thyroid Carcinoma Variants. Head Neck Pathol. 5(1), 51-56.



# All Code 

```{r quality_control_14_chips, eval=FALSE, include=FALSE}
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)

#read in .CEL files
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_rawdata")
data.thyroid = ReadAffy()
data.thyroid@cdfName <- "HGU133Plus2_Hs_ENST"

setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/rda")
save.image(file = "rawdata.thyroid.rda")

#quality control

#single chip control 
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/plots")
pdf(file="single_chip_control.pdf")
image(data.thyroid, col = rainbow(14, start = 0, end = 0.75)[10:1])
dev.off()
#alle chips scheinen okay 

#normalization
thyroid.vsnrma<-vsnrma(data.thyroid)
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/rda")
save.image(file = "normalized_data.rda")

#meanSdPlot
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/plots")
install.packages("hexbin")
library(hexbin)
pdf(file = 'meanSdPlot_thyroid_vsnrma_normalized.pdf')
meanSdPlot(thyroid.vsnrma)
dev.off()

#Boxplot
#before normalization
pdf(file = 'boxplot_thyroid_rawdata.pdf')
par(mar=c(9,5,3,5))
boxplot(data.thyroid,
        col=rainbow(14), 
        cex.axis=0.5, 
        ylab="intensity", 
        las=2, 
        main="Gene expression in pappillary thyroid cancer (GSE35570) \nbefore normalization (Handkiewicz-Junak et all., 2016)")
dev.off()

#after normalization
pdf(file = 'boxplot_thyroid_vsnrma_normalized.pdf')
par(mar=c(9,5,3,5))
boxplot(exprs(thyroid.vsnrma), 
        col=rainbow(14), 
        cex.axis=0.5, 
        ylab="log intensity", 
        las=2, 
        main="Gene expression in papillary thyroid cancer (GSE35570) \nafter vsnrma normalization (Handkiewicz-Junak et all., 2016)")
dev.off()

#density function 
#raw data
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/plots")
pdf(file = 'hist_thyroid_rawdata.pdf')
hist(data.thyroid, col=rainbow(14), main="Density function of log Intensity of papillary thyroid cancer (GSE35570) \nbefore normalization (Handkiewicz-Junak et all., 2016)")
dev.off()

#normalized data
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/plots")
pdf(file = 'hist_thyroid_vsnrma_normalised.pdf')
plot(density(exprs(thyroid.vsnrma)[,1]), type="n", xlab="log Intensity", ylim = c(0,1), main="Density function of log Intensity of papillary thyroid cancer (GSE35570) \nafter vsnrma normalization (Handkiewicz-Junak et all., 2016)")
for (i in 1:ncol(exprs(thyroid.vsnrma))) {
  lines(density(exprs(thyroid.vsnrma)[,i]), col=rainbow(14)[i])
}
dev.off()

#RNA degradation plot 
pdf(file = 'rnadeg_human_thyroid_cancer_rawdata.pdf')
rnadeg.raw = AffyRNAdeg(data.thyroid)
plotAffyRNAdeg(rnadeg.raw, col=rainbow(14))
title(sub="human thyroid cancer rawdata (GSE35570) (Handkiewicz-Junak et all., 2016)")
dev.off()

pdf(file = 'rnadeg_human_thyroid_cancer_rawdata_shifted.pdf')
plotAffyRNAdeg(rnadeg.raw, col=rainbow(14), transform="shift.only")
title(sub="human thyroid cancer rawdata (GSE35570) (Handkiewicz-Junak et all., 2016)")
dev.off()

#scatterplots
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/plots")
chipnames.CEL=colnames(exprs(thyroid.vsnrma))
chipnames = substr(chipnames.CEL, 1, nchar(chipnames.CEL)-4) #creating a vector of the chipnames without the .CEL
for (i in 1:13) {
  for (j in (i+1):14) {
    pdf(file = paste('scatterplot_thyroid_vsnrma_normalised', i, "_", j,'.pdf'))
    plot(exprs(thyroid.vsnrma)[,c(i,j)], pch=".", 
         main=paste("plot of probe", chipnames[i], "\nand", chipnames[j], "\n(Handkiewicz-Junak et all., 2016)"))
    abline(0,1,col="red")
    dev.off()
  }
}
```



```{r_quality_control_30_chips, eval=FALSE, include= FALSE}

#read in .CEL files
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_rawdata")
data.thyroid = ReadAffy()
data.thyroid@cdfName <- "HGU133Plus2_Hs_ENST"

setwd("/Users/Line/Documents/tra_project/with_30_chips/rda")
save.image(file = "rawdata.thyroid.rda")

#quality control

#single chip control 
setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
pdf(file="single_chip_control.pdf")
image(data.thyroid, col = rainbow(14, start = 0, end = 0.75)[10:1])
dev.off()
#alle chips scheinen okay 

#normalization
thyroid.vsnrma<-vsnrma(data.thyroid)
setwd("/Users/Line/Documents/tra_project/with_30_chips/rda")
save.image(file = "normalized_data.rda")

#meanSdPlot
setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
install.packages("hexbin")
library(hexbin)
pdf(file = 'meanSdPlot_thyroid_vsnrma_normalized.pdf')
meanSdPlot(thyroid.vsnrma)
dev.off()

#Boxplot
#before normalization
pdf(file = 'boxplot_thyroid_rawdata.pdf')
par(mar=c(9,5,3,5))
boxplot(data.thyroid,
        col=rainbow(14), 
        cex.axis=0.5, 
        ylab="intensity", 
        las=2, 
        main="Gene expression in pappillary thyroid cancer (GSE35570) \nbefore normalization (Handkiewicz-Junak et all., 2016)")
abline(h=c(6,8,10,12,14), lty=3)
boxplot(data.thyroid,
        col=rainbow(14), 
        cex.axis=0.5, 
        ylab="intensity", 
        las=2, 
        add=TRUE,
        main="Gene expression in pappillary thyroid cancer (GSE35570) \nbefore normalization (Handkiewicz-Junak et all., 2016)")
dev.off()

#after normalization
pdf(file = 'boxplot_thyroid_vsnrma_normalized.pdf')
par(mar=c(9,5,3,5))
boxplot(exprs(thyroid.vsnrma), 
        col=rainbow(14), 
        cex.axis=0.5, 
        ylab="log intensity", 
        las=2, 
        main="Gene expression in papillary thyroid cancer (GSE35570) \nafter vsnrma normalization (Handkiewicz-Junak et all., 2016)")
abline(h=c(6,8,10,12,14), lty=3)
boxplot(exprs(thyroid.vsnrma), 
        col=rainbow(14), 
        cex.axis=0.5, 
        ylab="log intensity", 
        las=2, 
        add=TRUE,
        main="Gene expression in papillary thyroid cancer (GSE35570) \nafter vsnrma normalization (Handkiewicz-Junak et all., 2016)")
dev.off()

#density function 
#raw data
setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
pdf(file = 'hist_thyroid_rawdata.pdf')
hist(data.thyroid, 
     col=rainbow(14), 
     main="Density function of log Intensity of papillary thyroid cancer (GSE35570) \nbefore normalization (Handkiewicz-Junak et all., 2016)")
abline(h=c(0.1,0.3,0.5), lty=3)
hist(data.thyroid, 
     col=rainbow(14), 
     add=TRUE,
     main="Density function of log Intensity of papillary thyroid cancer (GSE35570) \nbefore normalization (Handkiewicz-Junak et all., 2016)")
dev.off()

#normalized data
setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
pdf(file = 'hist_thyroid_vsnrma_normalised.pdf')
plot(density(exprs(thyroid.vsnrma)[,1]), 
     type="n", 
     xlab="log Intensity", 
     ylim = c(0,1), 
     main="Density function of log Intensity of papillary thyroid cancer (GSE35570) \nafter vsnrma normalization (Handkiewicz-Junak et all., 2016)")
for (i in 1:ncol(exprs(thyroid.vsnrma))) {
  lines(density(exprs(thyroid.vsnrma)[,i]), col=rainbow(14)[i])
}
abline(h=c(0.2,0.4,0.6,0.8), lty=3)
plot(density(exprs(thyroid.vsnrma)[,1]), 
     type="n", 
     xlab="log Intensity", 
     ylim = c(0,1), 
     main="Density function of log Intensity of papillary thyroid cancer (GSE35570) \nafter vsnrma normalization (Handkiewicz-Junak et all., 2016)")
for (i in 1:ncol(exprs(thyroid.vsnrma))) {
  lines(density(exprs(thyroid.vsnrma)[,i]), col=rainbow(14)[i])
}
dev.off()

#RNA degradation plot 
pdf(file = 'rnadeg_human_thyroid_cancer_rawdata.pdf')
rnadeg.raw = AffyRNAdeg(data.thyroid)
plotAffyRNAdeg(rnadeg.raw, col=rainbow(14))
title(sub="human thyroid cancer rawdata (GSE35570) (Handkiewicz-Junak et all., 2016)")
dev.off()

pdf(file = 'rnadeg_human_thyroid_cancer_rawdata_shifted.pdf')
plotAffyRNAdeg(rnadeg.raw, col=rainbow(14), transform="shift.only")
title(sub="human thyroid cancer rawdata (GSE35570) (Handkiewicz-Junak et all., 2016)")
dev.off()

#scatterplots
setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
chipnames.CEL=colnames(exprs(thyroid.vsnrma))
chipnames = substr(chipnames.CEL, 1, nchar(chipnames.CEL)-4) #creating a vector of the chipnames without the .CEL
for (i in 1:13) {
  for (j in (i+1):14) {
    pdf(file = paste('scatterplot_thyroid_vsnrma_normalised', i, "_", j,'.pdf'))
    plot(exprs(thyroid.vsnrma)[,c(i,j)], pch=".", 
         main=paste("plot of probe", chipnames[i], "\nand", chipnames[j], "\n(Handkiewicz-Junak et all., 2016)"))
    abline(0,1,col="red")
    dev.off()
  }
}
```



```{r Breast_Cancer_Data_Set, eval=FALSE, include=FALSE}

## Set working directory
wd <- "~/Users/macbook/Desktop/Bioinfo/Data Science Project"

# Load packages

library(affy)
library(vsn)
library(AnnotationDbi)

## Brainarray packages are located in .../Bioinfo/Data Science Project/

install.packages("~/Desktop/Bioinfo/Data Science Project/hgu133plus2hsenstprobe_25.0.0.tar.gz", repos = NULL, type = "source")
install.packages("~/Desktop/Bioinfo/Data Science Project/hgu133plus2hsenstcdf_25.0.0.tar.gz", repos = NULL, type = "source")

library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)

library(hexbin)

library(ggplot2)

## Read in CEL files, Breast Cancer

setwd(paste0(wd, "/rawdata-breast"))
cels <- list.files(pattern = "CEL")
dataRaw <- ReadAffy(filenames = cels, verbose = TRUE)
dataRaw@cdfName <- "HGU133Plus2_Hs_ENST"

new <- substr(cels, 1, nchar(cels)-4)
colnames(exprs(dataRaw)) <- new
rownames(dataRaw@phenoData) <- new
rownames(dataRaw@protocolData) <- new

save(dataRaw, file = paste0(wd, "/dataRaw.RData"))

####Microarray control and  Matrix fomation ####

## Single chip quality control
setwd(paste0(wd, "/plots"))
for (chipNo in seq_along(new)) { # for each chip
  pdf(file = paste0("QCs/QC_", new[chipNo], ".pdf")) # save in QCs/
  image(dataRaw[, chipNo], col = rainbow(100, start = 0, end = 0.75)[100:1]) # blue -> green -> red
  dev.off()
}

## Vsn normalisation

normdata <- vsnrma(dataRaw)
save(normdata, file = paste0(wd, "/norm.RData"))

## Mean SD plot
meanSdPlot(normdata)
ggsave("Mean_sd_plot.pdf")

## Extract expression values
dataexprs <- exprs(normdata)
head(dataexprs)
dim(dataexprs)
# 95721    10

## Ensemble IDs
dataexprs <- dataexprs[grepl("ENST", rownames(dataexprs)), ]
head(dataexprs)
dim(dataexprs)
# 95659    10

## Remove .x_at suffix
rownames(dataexprs) <- unlist(lapply(strsplit(rownames(dataexprs), split = "\\."), "[", 1))
head(dataexprs)

## Load ensemble IDs and gene symbols
setwd(wd)
ensembltxt <- read.csv("ensembl.103.txt", sep = ",")
head(ensembltxt)

transcriptIDs <- as.character(ensembltxt[, "Transcript.stable.ID"])
geneSymbol <- as.character(ensembltxt[, "HGNC.symbol"])
names(geneSymbol) <- transcriptIDs
head(geneSymbol)

## replace ensemble IDs with HGNC symbol
chipIDs <- rownames(dataexprs)
noMatchIDs <- chipIDs[!chipIDs %in% transcriptIDs] # transcript IDs that are not in ensemble_103.txt
length(noMatchIDs)
# 112 
newChipIDs <- chipIDs[chipIDs %in% transcriptIDs]
dataexprs <- dataexprs[newChipIDs, ]
dim(dataexprs)
# 95547    10

symbol <- geneSymbol[newChipIDs]
rownames(dataexprs) <- as.character(symbol)
head(dataexprs)

###Plots####

## Expression values before and after normalization

#Boxplot
#before normalization
setwd(paste0(wd, "/plots"))
pdf(file = 'boxplot_breast_cancer_rawdata.pdf')
par(mar=c(9,5,3,5))
boxplot(dataRaw,
        col=rainbow(14), 
        cex.axis=0.5, 
        ylab="intensity", 
        las=2, 
        main="Gene expression in Breast Cancer\nbefore normalization ")
dev.off()

#after normalization
setwd(paste0(wd, "/plots"))
pdf(file = 'boxplot_breastcancer_normalized.pdf')
par(mar=c(9,5,3,5))
boxplot(exprs(normdata), 
        col=rainbow(14), 
        cex.axis=0.5, 
        ylab="log intensity", 
        las=2, 
        main="Gene expression in Breast Cancer \nafter vsnrma normalization")
dev.off()

#density function 
#raw data
setwd(paste0(wd, "/plots"))
pdf(file = 'hist_breastcancer_rawdata.pdf')
hist(dataRaw, col=rainbow(14), main="Density function of log Intensity of papillary thyroid cancer (GSE35570) \nbefore normalization (Handkiewicz-Junak et all., 2016)")
dev.off()

#normalized data
setwd(paste0(wd, "/plots"))
pdf(file = 'hist_breastcancer_normalised.pdf')
plot(density(exprs(normdata)[,1]), type="n", xlab="log Intensity", ylim = c(0,1), main="Density function of log Intensity of papillary thyroid cancer (GSE35570) \nafter vsnrma normalization (Handkiewicz-Junak et all., 2016)")
for (i in 1:ncol(exprs(normdata))) {
  lines(density(exprs(normdata)[,i]), col=rainbow(14)[i])
}
dev.off()

#RNA degradation plot 
setwd(paste0(wd, "/plots"))
pdf(file = 'rnadeg_human_breast caner_rawdata.pdf')
rnadeg.raw = AffyRNAdeg(dataRaw)
plotAffyRNAdeg(rnadeg.raw, col=rainbow(14))
title(sub="human breast cancer rawdata ")
dev.off()

pdf(file = 'rnadeg_human_breastcancer_rawdata_shifted.pdf')
plotAffyRNAdeg(rnadeg.raw, col=rainbow(14), transform="shift.only")
title(sub="human breast cancer raw data")
dev.off()


####TRA List####

#set working directory

setwd(wd)

#load data

#data set 1

a=read.csv(file="tra.2014.mouse.5x.table.tsv",sep="\t")

tiss=a[,11]

ind=which(tiss=="thyroid")

TRA.symbol1=a[,3]

thyroid.TRA1=TRA.symbol1[ind]

str(a)

# data set 2 (no thyroid genes)

b=read.csv(file="tra.2014.mouse.4301.5x.table.tsv",sep="\t")

tiss2=b[,11]

ind2=which(tiss2=="Thyroid")

TRA.symbol2=b[,3]

thyroid.TRA2=TRA.symbol2[ind2]

#data set 3

c=read.csv(file="tra.2017.human.gtex.5x.table.tsv",sep="\t")

tiss3=c[,10]

ind3=which(tiss3=="Thyroid")

TRA.symbol3=c[,3]

thyroid.TRA3=TRA.symbol3[ind3]

#data set 4

d=read.csv(file="tra.2014.human.roth.5x.table.tsv",sep="\t")

tiss4=d[,11]

ind4=which(tiss4=="thyroid_gland")

TRA.symbol4=d[,3]

thyroid.TRA4=TRA.symbol4[ind4]

#data set 5

e=read.csv(file="tra.2014.human.5x.table.tsv",sep="\t")

tiss5=e[,11]

ind5=which(tiss5=="Thyroid")

TRA.symbol5=e[,3]

thyroid.TRA5=TRA.symbol5[ind5]


#unite the data 

thyroid.TRA=c(thyroid.TRA1,thyroid.TRA2,thyroid.TRA3,thyroid.TRA4,thyroid.TRA5)

thyroid.Trafinal=na.omit(thyroid.TRA)

y=as.data.frame(thyroid.Trafinal)

final_genes=y[!apply(y == "", 1, all), ]  

####Combining array data with TRA genes####

row.ind=which(final_genes%in%symbol)
dataexprs.sub=dataexprs[row.ind,]

#Box plot

order.vector=sort(colnames(t(dataexprs.sub)),index.return=T)$ix
setwd(paste0(wd, "/plots"))
pdf(file = 'boxplot_breastcancer_TRA.pdf')

boxplot(t(dataexprs.sub)[,order.vector],cex.axis=0.6,col=c("red","blue","yellow"),main="Gene expression of thyroid-specific TRAs in breast cancer",ylab="log2(gene expression)")
abline(h=c(6:12),col="grey",lty=3)

dev.off()
```



```{r all_code_(for_description), eval= FALSE, include=FALSE}

#extract expression values from normalized data 
thyroid.expr.matrix=exprs(thyroid.vsnrma)
head(thyroid.expr.matrix)
ensemble.id = rownames(thyroid.expr.matrix)
expr=substr(ensemble.id,0,15)
rownames(thyroid.expr.matrix)=expr

#read in annotation file 
setwd("/Users/Line/Documents/tra_project/thyroid_annotation_file")
a=read.csv("ensembl_103.txt",sep="\t")
head(a)
transcript.id.ensemble=as.character(a[,2])
symbol.ensembl=as.character(a[,4])
names(symbol.ensembl)=transcript.id.ensemble
head(symbol.ensembl)

#look into data matrix
head(thyroid.expr.matrix)
ensemble.id = rownames(thyroid.expr.matrix)
symbol=symbol.ensembl[ensemble.id]
head(symbol)
dim(thyroid.expr.matrix)
length(symbol)

#re-apply rownames to gene symbols in the data.matrix
rownames(thyroid.expr.matrix)=as.character(symbol)
head(thyroid.expr.matrix)

#save matrix as csv
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/tables")
write.csv(thyroid.expr.matrix, "thyroid_gene_expression")

##This is the matrix you can calculate with##

###extraction of genes of interest

#extract genes of interest
setwd("/Users/Line/Documents/tra_project/tra_tables")
b=read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv",sep=",")
tiss=b[,11]
ind=which(tiss=="thyroid")
TRA.symbol=b[,6]
thyroid.TRA1=TRA.symbol[ind]

c=read.csv(file="tra.2014.human.5x.table.tsv",sep="\t")
tiss=c[,11]
ind=which(tiss=="Thyroid")
TRA.symbol=c[,3]
thyroid.TRA2=TRA.symbol[ind]

d=read.csv(file="tra.2014.human.roth.5x.table.tsv",sep="\t")
tiss=d[,11]
ind=which(tiss=="thyroid_gland")
TRA.symbol=d[,3]
thyroid.TRA3=TRA.symbol[ind]

e=read.csv(file="tra.2014.mouse.5x.table.tsv",sep="\t")
tiss=e[,11]
ind=which(tiss=="thyroid")
TRA.symbol=e[,3]
thyroid.TRA4=TRA.symbol[ind]

f=read.csv(file="tra.2014.mouse.4301.5x.table.tsv",sep="\t")
tiss=f[,11]
ind=which(tiss=="thyroid")
TRA.symbol=f[,3]
thyroid.TRA5=TRA.symbol[ind]

g=read.csv(file="tra.2017.human.gtex.5x.table.tsv",sep="\t")
tiss=g[,10]
ind=which(tiss=="Thyroid")
TRA.symbol=g[,3]
thyroid.TRA6=TRA.symbol[ind]

thyroid.TRA = union(thyroid.TRA1, thyroid.TRA2)
thyroid.TRA = union(thyroid.TRA, thyroid.TRA3)
thyroid.TRA = union(thyroid.TRA, thyroid.TRA4)
thyroid.TRA = union(thyroid.TRA, thyroid.TRA6)

#sometimes upper case and lower case and other problems disturb the search (for example levels) 
#which you can eliminate with as.character and toupper

row.ind=which(toupper(thyroid.TRA) %in% as.character(symbol))
thyroid.expr.matrix.sub=thyroid.expr.matrix[row.ind,]

head(thyroid.expr.matrix.sub)
dim(thyroid.expr.matrix.sub)

#deleting the first 46 rows (AFFX...)
thyroid.expr.matrix.sub <- thyroid.expr.matrix.sub[-c(1:46),]
#reorder the matrix
thyroid.expr.matrix.sub <- thyroid.expr.matrix.sub[,c(1,5,6,7,8,2,3,4,9,10,11,12,13,14)]

#save matrix as csv
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/tables")
write.csv(thyroid.expr.matrix.sub, "thyroid_gene_expression_TRA")

#boxplot
par(las=2)
boxplot(t(thyroid.expr.matrix.sub),col=rainbow(length(rownames(thyroid.expr.matrix.sub))),main="gene expression of thyroid-specific genes in thyroid cancer",cex.axis=0.8)
#sort alphabetically
setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/plots")
pdf(file="boxplot_thyroid_cancer_genes_of_interest.pdf")

order.vector=sort(colnames(t(thyroid.expr.matrix.sub)),index.return=T)$ix
boxplot(t(thyroid.expr.matrix.sub)[,order.vector],col=rainbow(length(rownames(thyroid.expr.matrix.sub))),main="gene expression of thyroid-specific genes in papillary thyroid cancer \n(GSE35570) (Handkiewicz-Junak et all., 2016)",cex.axis=0.8)

dev.off()

#create three matrices (radiation_exposed/not_exposed/healthy)
radexp.matrix=thyroid.expr.matrix.sub[,c(1,5,6,7,8)]
radnonexp.matrix=thyroid.expr.matrix.sub[,c(2,3,4,9)]
healthy.matrix=thyroid.expr.matrix.sub[,c(10,11,12,13,14)]



#distribution our genes over chromosomes (out of human TRA lists=> List e and f can be ignored!)
#preparation => I need a matrix with the information "chromosome location" for each gene
#distribution our genes over chromosomes (out of human TRA lists=> List e and f can be ignored!)
#preparation => I need a matrix with the information "chromosome location" for each gene
tiss1=unique(as.character(b[,11])) # "thyroid"
th.genes1= b [b$Max_tissue %in% tiss1[31],]
dim(th.genes1)

tiss2= unique(as.character(c[,11])) 
length(tiss2) 
th.genes2= c[c$max.tissue %in% tiss2[35],  ]
dim(th.genes2)

tiss3=unique(as.character(d[,11])) #"thyroid_gland"
th.genes3= d[d$max.tissue %in% tiss3[20],]
dim(th.genes3)

tiss6=unique(as.character(g[,10])) 
th.genes6= g[g$max.tissue %in% tiss6[44],] # "Thyroid"
dim(th.genes6)

th.genes6[,11]= as.factor("no information") # we have no information about chromosome location of these genes

genes1= data.frame(th.genes1[,6],th.genes1[,2])
genes2= data.frame(th.genes2[,3],th.genes2[,7])
genes3= data.frame(th.genes3[,3],th.genes3[,7])
genes6= data.frame(th.genes6[,3],th.genes6[,11])

colnames(genes1)= c("gene.symbol", "chromosome")
colnames(genes2)= c("gene.symbol", "chromosome")
colnames(genes3)= c("gene.symbol", "chromosome")
colnames(genes6)= c("gene.symbol", "chromosome")

allgenesandchromosomes= rbind(genes1,genes2,genes3,genes6) # all genes of all TRA lists together
uniquegenes= duplicated(allgenesandchromosomes$gene.symbol) # selecting unique genes only
Index= which(uniquegenes==FALSE )
uniqueinfos=allgenesandchromosomes[Index,]

indexx=which(uniqueinfos$gene.symbol %in% symbol) # which genes are examined on our microarrays?
finalinfos= uniqueinfos[indexx,]
Ychrom= data.frame("inventedGene", "Y")
colnames(Ychrom)= c("gene.symbol", "chromosome")
finalinfos= rbind(finalinfos, Ychrom)
View(finalinfos) 

#final plots
chrnumbers= table(finalinfos[,2]) #wie ift ein chromosme vorkommt 
chromosomes= chrnumbers[c(2,13,17:23,3:12,14:16,26)]
chromswith0= c(chromosomes,chrnumbers[25])
chromswithY= c(chromosomes, chrnumbers[27])
chromswithY[24]= 0
View(chromosomes)

```




```{r chromosomeplots, eval=FALSE, include=FALSE}

library(viridisLite)
par(mai= c(2,2,2,2))
pdf(file="piechart_chromosomes.pdf")
pie(chromswith0, clockwise= TRUE,border= FALSE, col= c(plasma(23),"lightgoldenrodyellow"), init.angle=180, radius=1, main=" Distribution of \n thyroid specific genes \n over chromosomes \n Dinkelacker, 2019, PhD thesis, University of Heidelberg")
dev.off()
pdf(file="barplot_chromosomes.pdf")
par(las=1)
barplot(chromswithY, col=plasma(23), cex.names=0.5,main=" Distribution of thyroid specific genes over chromosomes \n Dinkelacker, 2019, PhD thesis, University of Heidelberg", xlab= "chromosomes", ylab= "number of genes", ylim=c(0,20))
abline(h=c(5,10,15,20), lty=3) # makes the grey lines. Tiping the barplot-code again, but with "add=TRUE" puts the lines in the background
barplot(chromswithY, col=plasma(23), add=TRUE, cex.names=0.5,main=" Distribution of thyroid specific genes over chromosomes \n Dinkelacker, 2019, PhD thesis, University of Heidelberg", xlab= "chromosomes", ylab= "number of genes", ylim=c(0,20))
dev.off()
```

  

```{r heatmap, eval=FALSE, include=FALSE, fig.cap= "Heatmap of 20 of our genes of interest. This heatmap illustrates the gene expression of the first 20 thyroid specific antigens in our matrix."}

library(pheatmap)

setwd("/Users/Line/Documents/tra_project/thyroid_cancer_sessions/plots")
pheatmap(thyroid.expr.matrix.sub, 
         main = "gene expression of thyroid TRA's \nin PTC and healthy thyroid tissue (GSE35570) \n(Handkiewicz-Junak et all., 2016)",
         cluster_rows = F, 
         cluster_cols = F, 
         cellheight = 3, 
         cellwidth = 3,
         legend =TRUE,
         fontsize = 3,
         filename = "pheatmap_thyroid_TRA_expression.pdf",
         gaps_col = c(5,9)
        )

```



```{r PCA_with_30_chips, eval=FALSE, include=FALSE}
#PCA of genes
library("ggplot2")
pca.t <- prcomp(t(thyroid.expr.matrix.sub), center = TRUE, scale = TRUE)
summary(pca.t) #with PC1 and PC2 40,87% of the variance are explained
col.pca.t = ifelse(colnames(thyroid.expr.matrix.sub) %in% colnames(radexp.matrix), 
                   "red", 
                   ifelse(colnames(thyroid.expr.matrix.sub) %in% colnames(radnonexp.matrix), 
                          "blue", 
                          "green"))
pdf(file="pca_genes")
plot(pca.t$x[,1], pca.t$x[,2], 
     col= col.pca.t,
     pch=16,
     xlab='PC1 (25,01 %)',ylab='PC2 (15,86 %)',
     main = "PC1 vs. PC2 for radiation-exposed PTC, \nnot-exposed PTC and healthy thyroid tissue \n(Handkiewicz-Junak et all., 2016)") 
legend("topright",c("radiation-exposed PTC","not-exposed PTC", "healthy"),fill=c("red","blue", "green"))

dev.off()

# Percentage of variance explained by dimensions
install.packages('factoextra')
library("factoextra")
eigenvalue <- round(get_eigenvalue(pca.t), 1)
variance.percent <- eigenvalue$variance.percent
head(eigenvalue)

#k-means 
km3 = kmeans(t(thyroid.expr.matrix.sub),centers=3,nstart = 100)
km3$cluster
cluster1.1 = c(which(km3$cluster==1))
cluster2.1 = c(which(km3$cluster==2))
cluster3.1 = c(which(km3$cluster==3))

library("factoextra")

setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
pdf(file="pca_genes_kmeans_3")
fviz_cluster(km3, data = t(thyroid.expr.matrix.sub),
             palette = c("#2E9FDF", "#00AFBB", "#E7B800"), 
             geom = "point",
             ellipse.type = "convex", 
             ggtheme = theme_bw()
)
dev.off()

km2 = kmeans(t(thyroid.expr.matrix.sub),centers=2,nstart = 100)
km2$cluster
cluster1.2 = c(which(km2$cluster==1))
cluster2.2 = c(which(km2$cluster==2))
cluster3.2 = c(which(km2$cluster==3))

pdf(file="pca_genes_kmeans_2")
fviz_cluster(km2, data = t(thyroid.expr.matrix.sub),
             palette = c("#2E9FDF", "#00AFBB", "#E7B800"), 
             geom = "point",
             ellipse.type = "convex", 
             ggtheme = theme_bw()
)
dev.off()
```



```{r anova_test_and_posthoc_plots, eval= FALSE, include=FALSE} 
### creating a new matrix for ANOVA test

groups= c("radexp","radexp","radexp","radexp","radexp","radexp","radexp","radexp","radexp","radexp",
          "radnonexp","radnonexp","radnonexp","radnonexp","radnonexp","radnonexp","radnonexp","radnonexp","radnonexp","radnonexp",
          "healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy")
anovamatrix=t(thyroid.expr.matrix.sub)
anovamatrix= cbind(groups,anovamatrix)

# anovamatrix[,1] = groups(exp, nonexp, healthy)
# anovamatrix[,i] = genes with their expression data


# ANOVA Test for each gene
sig.genes.vector=c()

for (i in 2:dim(anovamatrix)[2]){
  if (summary.aov(aov(anovamatrix[,i]~ anovamatrix[,1]))[[1]][[1,5]] < 0.05){
    sig.genes.vector = rbind(sig.genes.vector, colnames(anovamatrix)[i])
  }
  
}

sig.genes.matrix= thyroid.expr.matrix.sub[sig.genes.vector,]
# matrix with expression data of our significant diffent expressed genes

#install.packages("multcompView")
library(multcompView)

###preparation for boxplot
uniquegroups= unique(groups)

sig.anova.matrix = cbind(groups, t(sig.genes.matrix))  
# [i,] are the chips, [,i] the genes and [,1] are our groups
sig.genes.names= colnames(sig.anova.matrix[,-c(1)])

my_colors <- c( 
  "blue",
  "green", 
  "red"
) #defines colors

###The boxplots
setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")

for (i in 2:dim(sig.anova.matrix)[2]){
  pdf(file = paste('posthocgene', colnames(sig.anova.matrix)[i], ".pdf"))
  posthoc= TukeyHSD( aov(sig.anova.matrix[,i] ~ sig.anova.matrix[,1])) # makes posthoc for each gene
  
  Tukey.levels = posthoc[[1]][,4] # posthoc[[1]][,4] gibt uns die level an (a,b,c, oder ab)
  Tukey.labels = data.frame(multcompLetters(Tukey.levels)["Letters"])
  Tukey.labels$groups = rownames(Tukey.labels)
  Tukey.labels = Tukey.labels [order(Tukey.labels$groups), ] #defines the 3 groups, and labels, if group is significant different
  
  value= sig.anova.matrix[,i]
  data= data.frame(groups, value)
  data$value= as.numeric(data$value) 
  
  #boxplot coloured by group
  posthoc.plot= boxplot(data$value~data$groups, 
                        col=my_colors[as.factor(Tukey.labels[,1])],
                        xlab="tissue samples",
                        ylab= "expression values",
                        main= paste("posthoc of gene", sig.genes.names[i], "per group"))
  over <- 0.1*max( posthoc.plot$stats[nrow(posthoc.plot$stats),] )
  text( c(1:nlevels(as.factor(data$groups))) +0.5,
        posthoc.plot$stats[nrow(posthoc.plot$stats),], 
        Tukey.labels[,1], pos=1,
        col=my_colors[as.factor(Tukey.labels[,1])] ) #labels the boxes in the plot
  
  dev.off()
}
```



```{r PCA_and_kmeans_sick_only_code, eval=FALSE, include=FALSE}
#PCA of the sick patients only
pca.sick <- prcomp(t(thyroid.expr.matrix.sub)[1:20,], center = TRUE, scale = TRUE)
summary(pca.sick) #with PC1 and PC2 40,04% of the variance are explained
col.pca.sick = ifelse(colnames(thyroid.expr.matrix.sub) %in% colnames(radexp.matrix), "red", "blue")
setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
pdf(file="pca_sick")
plot(pca.sick$x[,1], pca.sick$x[,2], 
     col= col.pca.sick,
     pch=16,
     xlab='PC1 (26,06 %)',ylab='PC2 (13,99 %)',
     main = "PC1 vs. PC2 for radiation-exposed PTC \nand not-exposed PTC \n(Handkiewicz-Junak et all., 2016)") 
legend("bottomright",c("radiation-exposed PTC","not-exposed PTC"),fill=c("red","blue"))

dev.off()

#computing the best number of clusters 
#through the elbow method
wss.sick = sapply(1:7,function(k) { 
  kmeans(x=thyroid.expr.matrix.sub[,1:20], centers =k)$tot.withinss
})

plot(1:7,wss.sick,type='b',pch=19,xlab="Number of clusters K",
     ylab="Total within-clusters sum of squares") #3 seems to be the best number of clusters

#k-means of pca with sick patiets only
#with 2 clusters because of 2 groups
km.sick2 = kmeans(t(thyroid.expr.matrix.sub)[1:20,],centers=2,nstart = 100)
km.sick2$cluster

library("factoextra")

setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
pdf(file="pca_sick_kmeans_2")
fviz_cluster(km.sick2, data = t(thyroid.expr.matrix.sub)[1:20,],
             palette = c("#2E9FDF", "#00AFBB", "#E7B800"), 
             geom = "point",
             ellipse.type = "convex", 
             ggtheme = theme_bw()
)
dev.off()

#with 3 clusters because of elbow method
km.sick3 = kmeans(t(thyroid.expr.matrix.sub)[1:20,],centers=3,nstart = 100)
km.sick3$cluster

pdf(file="pca_sick_kmeans_3")
fviz_cluster(km.sick3, data = t(thyroid.expr.matrix.sub)[1:20,],
             palette = c("#2E9FDF", "#00AFBB", "#E7B800"), 
             geom = "point",
             ellipse.type = "convex", 
             ggtheme = theme_bw(),
             show.clust.cent = FALSE,
)
dev.off()
#3 clusters are a better fit
```



```{r ANOVA_sick_only, eval= FALSE, include=FALSE}
#seperating the chips according to the cancer subtype
cluster.sick.1 = c(which(km.sick3$cluster==1))
cluster.sick.2 = c(which(km.sick3$cluster==2))
cluster.sick.3 = c(which(km.sick3$cluster==3))

cluster.sick.1.matrix = thyroid.expr.matrix.sub[,cluster.sick.1]
cluster.sick.2.matrix = thyroid.expr.matrix.sub[,cluster.sick.2]
cluster.sick.3.matrix = thyroid.expr.matrix.sub[,cluster.sick.3]

cancer.subtype.matrix = cbind(cluster.sick.1.matrix, cluster.sick.2.matrix, cluster.sick.3.matrix)

cancer.groups= c("subtype 1","subtype 1","subtype 1","subtype 1","subtype 2","subtype 2","subtype 2","subtype 2","subtype 2",
                 "subtype 3","subtype 3","subtype 3","subtype 3","subtype 3","subtype 3","subtype 3","subtype 3","subtype 3",
                 "subtype 3","subtype 3")

cancer.anovamatrix =cbind(cancer.groups,t(cancer.subtype.matrix))

cancer.sig.genes.vector=c()

#reducing the number of genes by finding out which are significantly different expressed in the three subtypes
for (i in 2:dim(cancer.anovamatrix)[2]){
  if (summary.aov(aov(cancer.anovamatrix[,i]~ cancer.anovamatrix[,1]))[[1]][[1,5]] < 0.01){
    cancer.sig.genes.vector = rbind(cancer.sig.genes.vector, colnames(cancer.anovamatrix)[i])
  }
  
}

cancer.subtype.sig.genes.matrix= cancer.subtype.matrix[cancer.sig.genes.vector,]
# matrix with expression data of our significant different expressed genes between the three cancer subtypes

```



```{r evaluation_regression_model, eval=FALSE, include=FALSE}

#posthoc to find out which genes are significantly expressed in all 
#three subtypes to use those for the regression model
subtype.matrix = rbind(cancer.groups, cancer.subtype.sig.genes.matrix)
subtype.matrix = t(subtype.matrix)
modelgenes.sub = c()

library(multcompView)
for (i in 2:dim(subtype.matrix)[2]){
  posthoc= TukeyHSD( aov(subtype.matrix[,i] ~ subtype.matrix[,1]))
  Tukey.levels = posthoc[[1]][,4]
  Tukey.labels = data.frame(multcompLetters(Tukey.levels)["Letters"])
  if (Tukey.labels[1,1]== "a" & Tukey.labels[2,1]== "b" & Tukey.labels[3,1]== "c") {
    modelgenes.sub= c(modelgenes.sub, colnames(subtype.matrix)[i])}
}

modelgenes.sub

#control
which(colnames(subtype.matrix)=="SYPL1")

posthoc= TukeyHSD( aov(subtype.matrix[,22] ~ subtype.matrix[,1]))
Tukey.levels = posthoc[[1]][,4]
Tukey.labels = data.frame(multcompLetters(Tukey.levels)["Letters"])

Tukey.labels

#intsalling necessary packages for regression model
install.packages("nnet")
library("nnet")

#defining subtype 1 as reference category
subtype.matrix[,1]= relevel(factor(subtype.matrix[,1]), ref = "subtype 1")

#regeression model (training)
model.matrix.sub=subtype.matrix[,modelgenes.sub]
model.matrix.sub= cbind(cancer.groups,model.matrix.sub)

#Versuch Expressionswerte numerisch zu machen (bis jetzt ohne Erfolg)
model.matrix.sub=as.data.frame(model.matrix.sub)
for (i in 2:12){
  model.matrix.sub[,i]=apply(model.matrix.sub[,i,drop=F],2,as.numeric)
}

subtype.model=multinom(cancer.groups~.,
                       data = as.data.frame(model.matrix.sub[c(1,2,3,5,6,7,8,14,15,16,17,18,19,20),]))

#regressionmodel (prediction)
predict.subtype.1 = predict(subtype.model, model.matrix.sub[c(1,2,3,5,6,7,8,14,15,16,17,18,19,20),])
predict.subtype.2 = predict(subtype.model, model.matrix.sub[c(4,9,10,11,12,13),])    

#dioganal matrix if all subtypes were identified correctly 
tab1=table(predict.subtype.1, model.matrix.sub[c(1,2,3,5,6,7,8,14,15,16,17,18,19,20),1])
tab1 #is diagonal
tab2=table(predict.subtype.2, model.matrix.sub[c(4,9,10,11,12,13),1])
tab2 #is diagonal
#all subtypes are predicted correctly  
```



```{r big boxplot with 3 gropus, eval=FALSE, include=FALSE}
thyroid.expr.matrix.new=rbind(thyroid.expr.matrix.sub,thyroid.expr.matrix.sub,thyroid.expr.matrix.sub)

thyroid.expr.matrix.new[c(1:360),11:30]=NA
thyroid.expr.matrix.new[c(361:720),c(1:10,21:30)]=NA
thyroid.expr.matrix.new[c(721:1080),1:20]=NA

order.vector=sort(colnames(t(thyroid.expr.matrix.new)),index.return=T)$ix #gives position of genes in alphabetical order

setwd("/Users/Line/Documents/tra_project/with_30_chips/plots")
#genes starting with A
pdf(file = "boxplot_genes_of_interest_3groups_A")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[1:57]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
        )
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with B
pdf(file = "boxplot_genes_of_interest_3groups_B")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[58:84]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with C
pdf(file = "boxplot_genes_of_interest_3groups_C1")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[85:138]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

pdf(file = "boxplot_genes_of_interest_3groups_C2")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[139:192]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with D
pdf(file = "boxplot_genes_of_interest_3groups_D")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[193:237]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with E
pdf(file = "boxplot_genes_of_interest_3groups_E")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[238:270]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with F
pdf(file = "boxplot_genes_of_interest_3groups_F")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[271:324]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with G
pdf(file = "boxplot_genes_of_interest_3groups_G")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[325:366]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with H
pdf(file = "boxplot_genes_of_interest_3groups_H")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[367:402]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with I
pdf(file = "boxplot_genes_of_interest_3groups_I")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[403:429]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with J & K
pdf(file = "boxplot_genes_of_interest_3groups_J_and_K")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[430:459]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with L
pdf(file = "boxplot_genes_of_interest_3groups_L")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[460:489]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with M
pdf(file = "boxplot_genes_of_interest_3groups_M1")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[490:528]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

pdf(file = "boxplot_genes_of_interest_3groups_M2")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[529:567]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with N
pdf(file = "boxplot_genes_of_interest_3groups_N")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[568:606]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with O
pdf(file = "boxplot_genes_of_interest_3groups_O")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[607:636]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with P
pdf(file = "boxplot_genes_of_interest_3groups_P1")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[637:684]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

pdf(file = "boxplot_genes_of_interest_3groups_P2")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[685:729]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with Q and R
pdf(file = "boxplot_genes_of_interest_3groups_Q_and_R1")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[730:768]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with S
pdf(file = "boxplot_genes_of_interest_3groups_S1")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[808:855]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

pdf(file = "boxplot_genes_of_interest_3groups_S2")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[856:900]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with T 
pdf(file = "boxplot_genes_of_interest_3groups_T1")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[901:957]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

pdf(file = "boxplot_genes_of_interest_3groups_T2")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[958:1011]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with U and V
pdf(file = "boxplot_genes_of_interest_3groups_U_and_V")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[1012:1041]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()

#genes starting with X, Y and Z
pdf(file = "boxplot_genes_of_interest_3groups_X_Y_and_Z")
par(las=2)
boxplot(t(thyroid.expr.matrix.new)[,order.vector[1042:1080]],
        cex.axis=0.6,
        col=c("red","blue","green"),
        main="Gene expression of thyroid-specific TRAs in thyroid cancer, \nGSE35570 (Handkiewicz-Junak et all., 2016)",
        ylab="log2(gene expression)",
)
abline(h=c(6:12),lty=3)
legend("topright",c("radexp","nonradexp","norm"),col=c("red","blue","green"),pch=15,bg="white")
dev.off()
```



# Appendix

List of the 360 TRAs:
```{r appendix1, echo=FALSE}
list(rownames(thyroid.expr.matrix.sub))
```

List of the 153 significantly different expressed genes between radiation-exposed, not-exposed and healthy samples:
```{r appendix2, echo=FALSE}
list(rownames(sig.genes.matrix))
```

List of the 134 significantly different expressed genes between at least two of the three cancer subtypes:
```{r appendix3, echo=FALSE}
list(rownames(cancer.subtype.sig.genes.matrix))
```

List of the eleven significantly different expressed genes between all three cancer subtypes:
```{r appendix4, echo=FALSE}
list(modelgenes.sub)
```

