---
title: "Thyroid specific antigens DataScience"
author: "Camila Vacas, Line Weiß, Carina Keßler"
date: "7/2/2021"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 0. Gliederung Markdown:
1. Introduction 
    - biology 
    - our data set 
    - question we want to answer, and what we actually did to find the answers
    
2. Main part 

3. Results 

4. Conclusion 

5. Literature

6. Appendix 

## 1. Introduction
Tissue restricted antigens (TRA) are those which are expressed in at most five out of fourty-five tissues, and additionally in the lymph system, e.g. in the medullary thymic epithelial cells (mTECs), where the negative selection of the T-cells takes place. A defective regulation of TRA expression in mTECs can lead to autoimmune diseases, caused by the presence of T-cells reacting to a self-antigen. In most patients of e.g. the Hashimoto's or Grave's disease, the most common autoimmune disorders concerning the thyroid, are antibodies present against the thyroglobulin or the thyroid peroxidase. 
Due to this negative selection, the immune system cannot react properly to tumor cells which express tumor associated antigens (TAA), like differentiation or onco-fetal antigens (Klein,2006). The tumor escapes the immune system and can grow.

But also radiation has an positive impact on tumor development, especially on thyroid cancer: the thyroid is higly sensitive to long-term effects on radiation exposure. Though females tend to develope more often thyroid cancer than males, youth is a greater risk factor. Papillary carcinomas (PTC) represent with 80 % the most common thyroid tumor type (Albi, 2017, ganzer Absatz). Usually, there is one gene from the MAPK signal cascade mutated, the most frequently affected genes are *braf* (point mutation, mostly V600E) and *ret* (RET/PTC rearrangement).

We were working with 14 samples of thyroid tissues, in which five of them were extracted from patients, which were exposed to radiation and developped PTC, another four of sporadic PTC tissue and five samples from healthy patients. The gene sequencing was performed using total RNA and the gene chip **hgu133plus2hsenstcdf** (name? or is it the HSENS...).

After performing a (single chip) quality control we decided to use all fourteen chips for our analysis: after normalising the signal intensity of the chips, the gene expression was **equally? **distributed, the RNA degradation plot did not show any quality deficiencies, and as we scattered all chips against each other they showed the same pattern.

##  Quality Control (Plots)


```{r all code (for discription), include=FALSE}

library(affy)
library(vsn)
setwd("~/Desktop/Bioinfo/Data Science Project/rawdata/GSE35570 papillary thyroid cancer (PTC)")
data.thyroid = ReadAffy() #read in .CEL files
data.thyroid@cdfName <- "HGU133Plus2_Hs_ENST"

thyroid.vsnrma<-vsnrma(data.thyroid) #normalization
thyroid.expr.matrix=exprs(thyroid.vsnrma)
head(thyroid.expr.matrix)
ensemble.id = rownames(thyroid.expr.matrix)
expr=substr(ensemble.id,0,15)
rownames(thyroid.expr.matrix)=expr #extract expression values from normalized data 

#read in annotation file 
setwd("~/Desktop/Bioinfo/Data Science Project/TRA Daten")
a=read.csv("ensembl_103.txt",sep=",")
head(a)
transcript.id.ensemble=as.character(a[,2])
symbol.ensembl=as.character(a[,4])
names(symbol.ensembl)=transcript.id.ensemble
head(symbol.ensembl)

#look into data matrix
head(thyroid.expr.matrix)
ensemble.id = rownames(thyroid.expr.matrix)
symbol=symbol.ensembl[ensemble.id]
head(symbol)
dim(thyroid.expr.matrix)
length(symbol)

#re-apply rownames to gene symbols in the data.matrix
rownames(thyroid.expr.matrix)=as.character(symbol)

setwd("~/Desktop/Bioinfo/Data Science Project/TRA Daten")
b=read.csv(file="Human_protein_atlas_TRA_5median_genes_annotated.tsv",sep=",")
tiss=b[,11]
ind=which(tiss=="thyroid")
TRA.symbol=b[,6]
thyroid.TRA1=TRA.symbol[ind]

c=read.csv(file="tra.2014.human.5x.table.tsv",sep="\t")
tiss=c[,11]
ind=which(tiss=="Thyroid")
TRA.symbol=c[,3]
thyroid.TRA2=TRA.symbol[ind]

d=read.csv(file="tra.2014.human.roth.5x.table.tsv",sep="\t")
tiss=d[,11]
ind=which(tiss=="thyroid_gland")
TRA.symbol=d[,3]
thyroid.TRA3=TRA.symbol[ind]

e=read.csv(file="tra.2014.mouse.5x.table.tsv",sep="\t")
tiss=e[,11]
ind=which(tiss=="thyroid")
TRA.symbol=e[,3]
thyroid.TRA4=TRA.symbol[ind]

f=read.csv(file="tra.2014.mouse.4301.5x.table.tsv",sep="\t")
tiss=f[,11]
ind=which(tiss=="thyroid")
TRA.symbol=f[,3]
thyroid.TRA5=TRA.symbol[ind]

g=read.csv(file="tra.2017.human.gtex.5x.table.tsv",sep="\t")
tiss=g[,10]
ind=which(tiss=="Thyroid")
TRA.symbol=g[,3]
thyroid.TRA6=TRA.symbol[ind]

thyroid.TRA = union(thyroid.TRA1, thyroid.TRA2)
thyroid.TRA = union(thyroid.TRA, thyroid.TRA3)
thyroid.TRA = union(thyroid.TRA, thyroid.TRA4)
thyroid.TRA = union(thyroid.TRA, thyroid.TRA6)

#sometimes upper case and lower case and other problems disturb the search (for example levels) 
#which you can eliminate with as.character and toupper

row.ind=which(toupper(thyroid.TRA) %in% as.character(symbol))
thyroid.expr.matrix.sub=thyroid.expr.matrix[row.ind,]

head(thyroid.expr.matrix.sub)
dim(thyroid.expr.matrix.sub)

#deleting the first 46 rows (AFFX...)
thyroid.expr.matrix.sub <- thyroid.expr.matrix.sub[-c(1:46),]
#reorder the matrix
thyroid.expr.matrix.sub <- thyroid.expr.matrix.sub[,c(1,5,6,7,8,10,11,12,13,19,2,3,4,9,14:18,20,21:30)]

#save matrix as csv
setwd("~/Desktop/Bioinfo/Data Science Project/tables")
write.csv(thyroid.expr.matrix.sub, "thyroid_gene_expression_TRA")

#boxplot
par(las=2)
boxplot(t(thyroid.expr.matrix.sub),col=rainbow(length(rownames(thyroid.expr.matrix.sub))),main="gene expression of thyroid-specific genes in thyroid cancer",cex.axis=0.8)
#sort alphabetically
setwd("~/Desktop/Bioinfo/Data Science Project/plots")
pdf(file="boxplot_thyroid_cancer_genes_of_interest.pdf")

#create three matrices (radiation_exposed/not_exposed/healthy)
radexp.matrix=thyroid.expr.matrix.sub[,1:10]
radnonexp.matrix=thyroid.expr.matrix.sub[,11:20]
healthy.matrix=thyroid.expr.matrix.sub[,21:30]

#distribution our genes over chromosomes (out of human TRA lists=> List e and f can be ignored!)
#preparation => I need a matrix with the information "chromosome location" for each gene
tiss1=unique(as.character(b[,11])) # "thyroid"
th.genes1= b [b$Max_tissue %in% tiss1[31],]
dim(th.genes1)

tiss2= unique(as.character(c[,11])) 
length(tiss2) 
th.genes2= c[c$max.tissue %in% tiss2[35],  ]
dim(th.genes2)

tiss3=unique(as.character(d[,11])) #"thyroid_gland"
th.genes3= d[d$max.tissue %in% tiss3[20],]
dim(th.genes3)

tiss6=unique(as.character(g[,10])) 
th.genes6= g[g$max.tissue %in% tiss6[44],] # "Thyroid"
dim(th.genes6)

th.genes6[,11]= as.factor("no information") # we have no information about chromosome location of these genes

genes1= data.frame(th.genes1[,6],th.genes1[,2])
genes2= data.frame(th.genes2[,3],th.genes2[,7])
genes3= data.frame(th.genes3[,3],th.genes3[,7])
genes6= data.frame(th.genes6[,3],th.genes6[,11])

colnames(genes1)= c("gene.symbol", "chromosome")
colnames(genes2)= c("gene.symbol", "chromosome")
colnames(genes3)= c("gene.symbol", "chromosome")
colnames(genes6)= c("gene.symbol", "chromosome")

allgenesandchromosomes= rbind(genes1,genes2,genes3,genes6) # all genes of all TRA lists together
uniquegenes= duplicated(allgenesandchromosomes$gene.symbol) # selecting unique genes only
Index= which(uniquegenes==FALSE )
uniqueinfos=allgenesandchromosomes[Index,]

indexx=which(uniqueinfos$gene.symbol %in% symbol) # which genes are examined on our microarrays?
finalinfos= uniqueinfos[indexx,]
Ychrom= data.frame("inventedGene", "Y")
colnames(Ychrom)= c("gene.symbol", "chromosome")
finalinfos= rbind(finalinfos, Ychrom)
View(finalinfos) 

#final plots
chrnumbers= table(finalinfos[,2]) #wie ift ein chromosme vorkommt 
chromosomes= chrnumbers[c(2,13,17:23,3:12,14:16,26)]
chromswith0= c(chromosomes,chrnumbers[25])
chromswithY= c(chromosomes, chrnumbers[27])
chromswithY[24]= 0
View(chromosomes)

```

**Dieser Teil muss überarbeitet werden, je nachdem was wir jetzt wirklich machen**

In the next steps we focused on thyroid specific antigens and genes (TSG), which we determined by using five lists (from human and mouse samples) of genes and the tissues they express them the most.360 out of these genes were examined on the microarrays **(LIST SEE APENDIX????)**. We created a matrix including the expression data of all our genes of interest in the different samples (see below), so we can analyse them. We wanted to find out which of these thyroid specific genes are differentially expressed in thyroid papillarys carcinomas (induced or not induced by radiation) and whether there is a relation between some of them or not. We wanted to cluster them and try to predict the expression of a specific gene based on the expression data of another one.

```{r head(thyroid.expr.matrix.sub), echo=TRUE}
head(thyroid.expr.matrix.sub[,c(1,2,6,7,9,11:13,20:25)])
#the first 6 TSG and their expression levels in cancer tissue exposed to radiation, 
#sporadic cancer tissue and healthy tissue
```

## 2. Main Part
  
  We started our project by taking a closer look at the data. We found out, that the TSG are spread over all human chromosomes, except for the Y chromosome. Unfortunately we cannot say on which chromosomes can be found the most TSG, because of the missing information almost half of our genes. 


```{r chromosome plot, echo=FALSE}
library(viridisLite)
par(las=1)
barplot(chromswithY, col=plasma(23), main=" Distribution of thyroid specific genes over chromosomes \n Dinkelacker, 2019, PhD thesis, University of Heidelberg", xlab= "chromosomes", ylab= "number of genes", ylim=c(0,20))
abline(h=c(5,10,15,20), lty=3) # makes the grey lines. Tiping the barplot-code again, but with "add=TRUE" puts the lines in the background
barplot(chromswithY, col=plasma(23), add=TRUE, main=" Distribution of thyroid specific genes over chromosomes \n Dinkelacker, 2019, PhD thesis, University of Heidelberg", xlab= "chromosomes", ylab= "number of genes", ylim=c(0,20))

```

A heatmap revealed that the relative expression was from 5 to 11, 11 beeing highly expressed. Most of the genes showed an expression between 5 to 7.

```{r heat map, echo=FALSE}
littlematrix= thyroid.expr.matrix.sub[1:20,c(1,2,6,7,9,11:13,20:25)]
library(pheatmap)
heat=pheatmap(littlematrix, 
                             main = "gene expression of thyroid TRA's \nin PTC and healthy thyroid tissue (GSE35570) \n(Handkiewicz-Junak et all., 2016)",
                             cluster_rows = F, 
                             cluster_cols = F, 
                             cellheight = 3, 
                             cellwidth = 3,
                             legend =TRUE,
                             fontsize = 3,
                             filename = "pheatmap_thyroid_TRA_expression.pdf",
                             gaps_col = c(5,9)
               )
heat
```
Next, we did a principal component analysis (PCA) and coloured the different spots according to the tissue sample the microarray was prepared with. The five healthy samples and the five radiation exposed samples formed both a clear individual cluster. Two of the samples of the sick patients, who were not exposed to radiation, formed a cluster, while the other two seemed to be more similiar to the radiation exposed samples 
[Bild 1]: /Users/Line/Documents/tra_project/thyroid_cancer_sessions/plots/pca_chips. 
![Alt-Text][Bild 1]
To find out, wether those were just outliers, we repeated the principal component analysis with 30 chips, 10 chips in each group. This time it was clear, that the healthy samples formed a cluster and the sick samples another one. 
```{r}
#PCA with 30 chips
pca.t <- prcomp(t(thyroid.expr.matrix.sub), center = TRUE, scale = TRUE)
summary(pca.t) #with PC1 and PC2 40,87% of the variance are explained
col.pca.t = ifelse(colnames(thyroid.expr.matrix.sub) %in% colnames(radexp.matrix), 
                   "red", 
                   ifelse(colnames(thyroid.expr.matrix.sub) %in% colnames(radnonexp.matrix), 
                          "blue", 
                          "green"))

plot(pca.t$x[,1], pca.t$x[,2], 
     col= col.pca.t,
     pch=16,
     xlab='PC1',ylab='PC2',
     main = "PC1 vs. PC2 for radiation-exposed PTC (red), \nnot-exposed PTC (blue) and 
     healthy thyroid tissue (green) \n(Handkiewicz-Junak et all., 2016)") 

# shows all 30 chips, shows that healthy and cancer tissues are different expressed


# Percentage of variance explained by dimensions
#install.packages('factoextra')
library("factoextra")
eigenvalue <- round(get_eigenvalue(pca.t), 1)
variance.percent <- eigenvalue$variance.percent
head(eigenvalue)


#legende einfügen
#Varianz an die Achsen schreiben

#k-means 
pca.matrix<-pca.t[["rotation"]]
km3 = kmeans(t(thyroid.expr.matrix.sub),centers=3,nstart = 100)
km3$cluster
cluster1.1 = c(which(km3$cluster==1))
cluster2.1 = c(which(km3$cluster==2))
cluster3.1 = c(which(km3$cluster==3))

library("factoextra")


fviz_cluster(km3, data = t(thyroid.expr.matrix.sub),
             palette = c("#2E9FDF", "#00AFBB", "#E7B800"), 
             geom = "point",
             ellipse.type = "convex", 
             ggtheme = theme_bw()
)

# "same plot" as above, but with defined clusters, and cluster borders

km2 = kmeans(t(thyroid.expr.matrix.sub),centers=2,nstart = 100)
km2$cluster
cluster1.2 = c(which(km2$cluster==1))
cluster2.2 = c(which(km2$cluster==2))
cluster3.2 = c(which(km2$cluster==3))


fviz_cluster(km2, data = t(thyroid.expr.matrix.sub),
             palette = c("#2E9FDF", "#00AFBB", "#E7B800"), 
             geom = "point",
             ellipse.type = "convex", 
             ggtheme = theme_bw()
)

# shows that healthy and cancer tissue form 2 seperated cluster


```
To investigate, which of the 360 TRA's showed a significant change in gene expression upon the three different groups, we performed an ANOVA test for each gene and stored the 153 significantly different expressed ones in a new matrix. 
```{r anova.groups, echo=FALSE} 
### creating a new matrix for ANOVA test

groups= c("radexp","radexp","radexp","radexp","radexp","radexp","radexp","radexp",
          "radexp","radexp",
          "radnonexp","radnonexp","radnonexp","radnonexp","radnonexp","radnonexp",
          "radnonexp","radnonexp","radnonexp","radnonexp",
          "healthy","healthy","healthy","healthy","healthy","healthy","healthy",
          "healthy","healthy","healthy")

anovamatrix=t(thyroid.expr.matrix.sub)
anovamatrix= cbind(groups,anovamatrix)

# anovamatrix[,1] = groups(exp, nonexp, healthy)
# anovamatrix[,i] = genes with their expression data


#anovamatrix = t(thyroid.expr.matrix.sub) with an extra column that specifies the corresponding group of each chip
#**or** anovamatrix[1:3,]

```

``` {r anova, echo=TRUE}

# ANOVA Test for each gene

sig.genes.vector=c()
for (i in 2:dim(anovamatrix)[2]){
  if (summary.aov(aov(anovamatrix[,i]~ anovamatrix[,1]))[[1]][[1,5]] < 0.05){
    sig.genes.vector = rbind(sig.genes.vector, colnames(anovamatrix)[i])
  }
  
}

sig.genes.matrix= thyroid.expr.matrix.sub[sig.genes.vector,]
# when the difference in gene expression for gene i between two groups is higher
# than 5%, the corresponding gene name (=> colnames(anovamatrix)[i]) is added in
# the vector sig.genes.vector. 
#The sig.genes.matrix contains the expression values of these genes.
```

A posthoc test of each of the significantly different expressed genes showed us how the acccording gene was expressed in the three groups and between which of the groups there was the significant difference.

**Ich würde hier exemplarisch einen Plot zeigen, welches Gen sieht denn gut aus?**


```{r posthoc, echo=FALSE}
#for-loop posthoc for every gene
#install.packages("multcompView")
library(multcompView)
uniquegroups= unique(groups)

sig.anova.matrix = cbind(groups, t(sig.genes.matrix))  
# [i,] are the chips, [,i] the genes and [,1] are our groups
sig.genes.names= colnames(sig.anova.matrix[,-c(1)])

my_colors <- c( 
  "blue",
  "green", 
  "red"
) #defines colors

###The boxplots
setwd("~/Desktop/Bioinfo/Data Science Project/plots")
for (i in 2:dim(sig.anova.matrix)[2]){
  pdf(file = paste('posthocgene', colnames(sig.anova.matrix)[i], ".pdf"))
  posthoc= TukeyHSD( aov(sig.anova.matrix[,i] ~ sig.anova.matrix[,1])) # makes posthoc for each gene
  
  Tukey.levels = posthoc[[1]][,4]
  Tukey.labels = data.frame(multcompLetters(Tukey.levels)["Letters"])
  Tukey.labels$groups = rownames(Tukey.labels)
  Tukey.labels = Tukey.labels [order(Tukey.labels$groups), ] #defines the 3 groups, and labels, if group is significant different
  
  value= sig.anova.matrix[,i]
  data= data.frame(groups, value)
  data$value= as.numeric(data$value) 
  
  #boxplot coloured by group
  posthoc.plot= boxplot(data$value~data$groups, 
                        col=my_colors[as.factor(Tukey.labels[,1])],
                        xlab="tissue samples",
                        ylab= "expression values",
                        main= paste("posthoc of gene", sig.genes.names[i], "per group"))
  over <- 0.1*max( posthoc.plot$stats[nrow(posthoc.plot$stats),] )
  text( c(1:nlevels(as.factor(data$groups))) +0.5,
        posthoc.plot$stats[nrow(posthoc.plot$stats),], 
        Tukey.labels[,1], pos=1,
        col=my_colors[as.factor(Tukey.labels[,1])] ) #labels the boxes in the plot
  
  dev.off()
}
```

**You can see here, that the gene XXX is XXX expressed in XXX.**

While doing the PCA, we found out, that the total expression of radiation exposed and non exposed tissue did not differ as much as we thought it would. So we concentrated on the cancer tissue and performed another PCA and a k-means clustering. As expected, the chips could not be seperated in two groups according to their radiation exposure. 
The elbow method,we used to define the perfect number of clusters, revealed, that the cancer tissues should be divided into three groups, illustrated below.

```{r PCA sick only, echo=FALSE}
km.sick3 = kmeans(t(thyroid.expr.matrix.sub)[1:20,],centers=3,nstart = 100)
km.sick3$cluster
fviz_cluster(km.sick3, data = t(thyroid.expr.matrix.sub)[1:20,],
             palette = c("#2E9FDF", "#00AFBB", "#E7B800"), 
             geom = "point",
             ellipse.type = "convex",
             show.clust.cent = FALSE,
             ggtheme = theme_bw()
)
```
Next, we examined the gene expression pattern of these three cancer subtypes. Again, we performed an ANOVA test, defining the genes, that are significantly different expressed in one of our subtypes. We found 134. 

```{r foldchange, echo=TRUE}
cluster.sick.1 = c(which(km.sick3$cluster==1))
cluster.sick.2 = c(which(km.sick3$cluster==2))
cluster.sick.3 = c(which(km.sick3$cluster==3))

cluster.sick.1.matrix = thyroid.expr.matrix.sub[,cluster.sick.1]
cluster.sick.2.matrix = thyroid.expr.matrix.sub[,cluster.sick.2]
cluster.sick.3.matrix = thyroid.expr.matrix.sub[,cluster.sick.3]

cancer.subtype.matrix = cbind(cluster.sick.1.matrix, cluster.sick.2.matrix, cluster.sick.3.matrix)

cancer.groups= c("subtype 1","subtype 1","subtype 1","subtype 1","subtype 2",
                 "subtype 2","subtype 2","subtype 2","subtype 2",
                 "subtype 3","subtype 3","subtype 3","subtype 3","subtype 3",
                 "subtype 3","subtype 3","subtype 3","subtype 3",
                 "subtype 3","subtype 3")

cancer.anovamatrix =cbind(cancer.groups,t(cancer.subtype.matrix))

cancer.sig.genes.vector=c()

# reducing the number of genes by finding out which are significantly different 
# expressed in the three subtypes

for (i in 2:dim(cancer.anovamatrix)[2]){
  if (summary.aov(aov(cancer.anovamatrix[,i]~ cancer.anovamatrix[,1]))[[1]][[1,5]] < 0.01){
    cancer.sig.genes.vector = rbind(cancer.sig.genes.vector, colnames(cancer.anovamatrix)[i])
  }
  
}

cancer.subtype.sig.genes.matrix= cancer.subtype.matrix[cancer.sig.genes.vector,]
# matrix with expression data of our significant different expressed genes between
# the three cancer subtypes
```

As we performed the vs-normalisation right at the beginning, the fold change is the absolute difference in gene expression, and we do not need to use the logarithm. We defined 8 as our treshold and continued with the genes, that are expressed at least eight times higher in one subtype than in another one.


With these **how many at the end?** genes we performed a multinominal regression analysis. We wanted to predict, due to the expression pattern based on these genes, the cancer subtype. **For the model we used 6 of our 20 cancer tissue chips.**

```{r regression analysis sick, echo=TRUE}
#posthoc to find out which genes are significantly expressed in all 
#three subtypes to use those for the regression model
subtype.matrix = rbind(cancer.groups, cancer.subtype.sig.genes.matrix)
subtype.matrix = t(subtype.matrix)
modelgenes.sub = c()

library(multcompView)
for (i in 2:dim(subtype.matrix)[2]){
  posthoc= TukeyHSD( aov(subtype.matrix[,i] ~ subtype.matrix[,1]))
  Tukey.levels = posthoc[[1]][,4]
  Tukey.labels = data.frame(multcompLetters(Tukey.levels)["Letters"])
  if (Tukey.labels[1,1]== "a" & Tukey.labels[2,1]== "b" & Tukey.labels[3,1]== "c") {
    modelgenes.sub= c(modelgenes.sub, colnames(subtype.matrix)[i])}
}

modelgenes.sub

#control
which(colnames(subtype.matrix)=="SYPL1")



```

**diese Tabelle zeigt, dass alle Chips richitg zugeordnet werden. Bzw. die Tabelle würde ich vielleicht eher bei den anderen Chips zeigen, um zu demonstrieren wie gut unser Modell ist.**

## 3. Results: 


## 4. Conclusion:


## 5. Literature: 

TRA datas:
GTEX 2015, Human genomics. The Genotype-Tissue Expression (GTEx) pilot analysis: multitissue gene regulation in humans. GTEx Consortium. Science. 2015 May 8;348(6235):648-60. https://pubmed.ncbi.nlm.nih.gov/25954001/

Lattin et al. 2008, Expression analysis of G Protein-Coupled Receptors in mouse macrophages. Immunome Res. 2008 Apr 29;4:5. https://pubmed.ncbi.nlm.nih.gov/18442421/

Roth et al. 2006, Gene expression analyses reveal molecular relationships among 20 regions of the human CNS. Neurogenetics. 2006 May;7(2):67-80. https://pubmed.ncbi.nlm.nih.gov/16572319/

Su et al. 2002, Large-scale analysis of the human and mouse transcriptomes. Proc Natl Acad Sci U S A. 2002 Apr 2;99(7):4465-70. https://pubmed.ncbi.nlm.nih.gov/11904358/

Su et al. 2004, A gene atlas of the mouse and human protein-encoding transcriptomes. Proc Natl Acad Sci U S A. 2004 Apr 20;101(16):6062-7. https://pubmed.ncbi.nlm.nih.gov/15075390/

Uhlén et al. 2015, Proteomics. Tissue-based map of the human proteome. Science. 2015 Jan 23;347(6220):1260419. https://pubmed.ncbi.nlm.nih.gov/25613900/


in generell:
Albi et al. 2017. Radiation and Thyroid Cancer. Int J Mol Sci. 2017 Apr 26;18(5):911. https://pubmed.ncbi.nlm.nih.gov/28445397/
